distributed/security.py

```diff
@@ -10,11 +11,70 @@
 import dask
 
 
-__all__ = ("Security",)
+__all__ = ("Security", "TLSSecurity")
 
 
-class Security(object):
-    """Security configuration for a Dask cluster.
+class Security(ABC):
+    """
+    Security configuration for a Dask cluster
+    """
+
+    __slots__ = ()
+
+    def __init__(self, **kwargs):
+        extra = set(kwargs).difference(self.__slots__)
+        if extra:
+            raise TypeError("Unknown parameters: %r" % sorted(extra))
+
+    @abstractmethod
+    def get_connection_args(self, role: str):
+        """
+        Get the *connection_args* argument for a connect() call with
+        the given *role*.
+        :type role: str: one of client, scheduler, worker
+        """
+        pass
+
+    @abstractmethod
+    def get_listen_args(self, role: str):
+        """
+        Get the *connection_args* argument for a listen() call with
+        the given *role*.
+        :type role: str: one of client, scheduler, worker
+        """
+        pass
+
+    def _set_field(self, kwargs: dict, field: str, config_name: str):
+        if field in kwargs:
+            out = kwargs[field]
+        else:
+            out = dask.config.get(config_name)
+        setattr(self, field, out)
+
+    def __repr__(self):
+        keys = sorted(self.__slots__)
+        items = []
+        for k in keys:
+            val = getattr(self, k)
+            if val is not None:
+                if isinstance(val, str) and "\n" in val:
+                    items.append((k, "..."))
+                else:
+                    items.append((k, repr(val)))
+        return (
+            type(self).__name__
+            + "("
+            + ", ".join("%s=%s" % (k, v) for k, v in items)
+            + ")"
+        )
+
+    @classmethod
+    def temporary(cls):
+        return TLSSecurity.temporary()
```

Comment:
This does not make sense in the new base class, but to reduce the breakage for out-of-tree codebases I added it here.

If we have a `deprecated` decorator, this would be a place to add it.